pkg/apis/core/validation/validation.go

```diff
@@ -3449,7 +3449,7 @@ func ValidatePodSpec(spec *core.PodSpec, podMeta *metav1.ObjectMeta, fldPath *fi
 	}
 
 	if len(spec.Hostname) > 0 {
-		allErrs = append(allErrs, ValidateDNS1123Label(spec.Hostname, fldPath.Child("hostname"))...)
+		allErrs = append(allErrs, ValidateDNS1123Subdomain(spec.Hostname, fldPath.Child("hostname"))...)
```

Comment:
I'll defer to @thockin / networking folks on whether this change makes sense or not, but it is unclear to me if this is correct... if this field is allowed to contain multiple segments, what implications does that have for the peer `subdomain` field?

```
	// Specifies the hostname of the Pod
	// If not specified, the pod's hostname will be set to a system-defined value.
	// +optional
	Hostname string `json:"hostname,omitempty" protobuf:"bytes,16,opt,name=hostname"`
	// If specified, the fully qualified Pod hostname will be "<hostname>.<subdomain>.<pod namespace>.svc.<cluster domain>".
	// If not specified, the pod will not have a domainname at all.
	// +optional
```

Reply:
if the change does end up getting blessed by networking folks, relaxing validation has to be done over two releases so that we don't allow data to be persisted that is considered invalid by the previous version of the API server.

Reply:
indeed, I think that this is very risky https://github.com/kubernetes/kubernetes/issues/104195#issuecomment-908344150